Make intra-highlights customizable

obcat opened the pull request:
Hi @airblade.

Currently, the user cannot customize the color of the intra-highlights, "GitGutterAddIntraLine" and "GitGutterDeleteIntraLine", because we call the ":highlight" commands without "default" argument (:help :highlight-default). So I suggest adding it. What do you think?

Thank you.

airblade commented:
That would be fine.

Please could you also update the docs?

- the help doc in the [highlight](https://github.com/airblade/vim-gitgutter/blob/7d9b7125ff2533f6c520573f54e91db2f13a44b0/doc/gitgutter.txt#L571) section
- the readme, in a new section after the [line number highlights](https://github.com/airblade/vim-gitgutter#line-number-highlights) section

obcat commented:
Thank you for the positive response!

> Please could you also update the docs?
> 
> * the help doc in the [highlight](https://github.com/airblade/vim-gitgutter/blob/7d9b7125ff2533f6c520573f54e91db2f13a44b0/doc/gitgutter.txt#L571) section
> * the readme, in a new section after the [line number highlights](https://github.com/airblade/vim-gitgutter#line-number-highlights) section

Sure, please give me some time.

obcat commented:
I've updated the docs.

I also added a way to customize the diff syntax colours as well as intra-line highlighting, because I feel it a bit unnatural that the latter is only documented.

airblade commented:
Thanks!

obcat commented:
Yey! Thanks for the quick review and merge!
